src/tap/src/tapcell.cpp

```diff
@@ -33,41 +33,1306 @@
+          incnr_master = incnrcap_nwout_master;
+          west_ori = odb::dbOrientType::R180;
+        }
+
+        // NE corner
+        if (checkSymmetry(incnr_master, top_row_ori)) {
+          buildCell(block, incnr_master, top_row_ori, x_end, top_row_y, prefix);
+        }
+        // NW corner
+        if (checkSymmetry(incnr_master, west_ori)) {
+          buildCell(block,
+                    incnr_master,
+                    west_ori,
+                    (x_start - incnr_master->getWidth()),
+                    top_row_y,
+                    prefix);
+        }
+      }
+
+      if (bot_row >= 1) {
+        odb::dbRow* bot_row_inst = rows[bot_row].front();
+        odb::dbOrientType bot_row_ori = bot_row_inst->getOrient();
+        odb::Rect rowbb1;
+        bot_row_inst->getBBox(rowbb1);
+        int bot_row_y = rowbb1.yMin();
+
+        row_start = x_start;
+        row_end = x_end;
+        if (row_start == -1) {
+          row_start = rowbb1.xMin();
+          row_start = row_start + corner_cell_width;
+        }
+        if (row_end == -1) {
+          odb::Rect rowbb3;
+          rows[bot_row].back()->getBBox(rowbb3);
+          row_end = rowbb3.xMax();
+          row_end = row_end - corner_cell_width;
+        }
+
+        // do bottom row
+        insertAtTopBottomHelper(block,
+                                0,
+                                1,
+                                bot_row_ori,
+                                row_start,
+                                row_end,
+                                bot_row_y,
+                                tap_nwintie_master,
+                                tap_nwin2_master,
+                                tap_nwin3_master,
+                                tap_nwouttie_master,
+                                tap_nwout2_master,
+                                tap_nwout3_master,
+                                prefix);
+
+        // do corners
+        if (bot_row_ori == odb::dbOrientType::MX) {
+          incnr_master = incnrcap_nwin_master;
+          west_ori = odb::dbOrientType::R180;
+        } else {
+          incnr_master = incnrcap_nwout_master;
+          west_ori = odb::dbOrientType::MY;
+        }
+
+        // SE corner
+        if (checkSymmetry(incnr_master, bot_row_ori)) {
+          buildCell(block, incnr_master, bot_row_ori, x_end, bot_row_y, prefix);
+        }
+        // SW corner
+        if (checkSymmetry(incnr_master, west_ori)) {
+          buildCell(block,
+                    incnr_master,
+                    west_ori,
+                    (x_start - incnr_master->getWidth()),
+                    bot_row_y,
+                    prefix);
+        }
+      }
+    }
+  }
+  int blkgs_cnt = phy_idx_ - start_phy_idx;
+  logger_->info(utl::TAP, 7, "Inserted {} cells near blockages.", blkgs_cnt);
+  return blkgs_cnt;
+}
+
+std::pair<int, int> Tapcell::getMinMaxX(vector<vector<odb::dbRow*>>& rows)
+{
+  int min_x = -1;
+  int max_x = -1;
```

Comment:
Just initialize these with numeric_limits<int>::max() and min() and use std::max() min() below without any if statements.